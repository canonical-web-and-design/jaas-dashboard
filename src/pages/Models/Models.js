import React from "react";
import TableList from "components/TableList/TableList";

export default function Models() {
  return (
    <div className="p-strip">
      <div className="row">
        <TableList />
      </div>
    </div>
  );
}
