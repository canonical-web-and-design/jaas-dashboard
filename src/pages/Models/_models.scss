@import "vanilla-framework/scss/vanilla";

.models {
  &__count {
    display: none;
    font-size: 0.875rem;
    padding: 0 1.5rem;

    @media (min-width: $breakpoint-large) {
      display: block;
    }
  }

  &__header {
    align-items: center;
    display: grid;
    grid-template-columns: 1fr;
    padding: 0.575rem;

    @media (min-width: $breakpoint-medium) {
      grid-template-columns: 1fr 280px;
    }

    @media (min-width: $breakpoint-large) {
      grid-template-columns: repeat(2, 1fr) 260px;
    }
  }

  // Tooltips override
  .p-tooltip--top-center .p-tooltip__message {
    left: calc(50% - 0.875rem);
    margin-bottom: -0.5rem;
  }
}
