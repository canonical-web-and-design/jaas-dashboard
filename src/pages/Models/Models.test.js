import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { mount } from "enzyme";
import { MemoryRouter, Router } from "react-router";
import dataDump from "testing/complete-redux-store-dump";
import { createMemoryHistory } from "history";

import Models from "./Models";

const mockStore = configureStore([]);

describe("Models page", () => {
  it("renders without crashing", () => {
    const store = mockStore(dataDump);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Models />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find("Header"));
    expect(wrapper.find("TableList"));
  });

  it("has a header which shows the model counts", () => {
    const store = mockStore(dataDump);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Models />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find(".models__count").text()).toBe(
      "17 models: 6 blocked, 4 alerts, 7 running"
    );
  });

  it("displays correct grouping view", () => {
    const history = createMemoryHistory();
    const store = mockStore(dataDump);
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <Models />
        </Router>
      </Provider>
    );
    expect(
      wrapper.find(".p-model-group-toggle__button.is-selected").text()
    ).toBe("status");
    wrapper.find("button[value='owner']").simulate("click", {
      target: { value: "owner" }
    });
    expect(
      wrapper.find(".p-model-group-toggle__button.is-selected").text()
    ).toBe("owner");
    const searchParams = new URLSearchParams(history.location.search);
    expect(searchParams.get("groupedby")).toEqual("owner");
    expect(wrapper.find(".owners-group"));
  });
});
