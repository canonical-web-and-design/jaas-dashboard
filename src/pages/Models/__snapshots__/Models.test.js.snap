// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Models page renders without crashing and matches snapshot 1`] = `
<div
  className="p-strip"
>
  <div
    className="row"
  >
    <TableList />
  </div>
</div>
`;
